src/Storages/MergeTree/MergeTreeSink.cpp

```diff
@@ -74,6 +108,29 @@ void MergeTreeSink::consume(Chunk chunk)
 
     for (auto & current_block : part_blocks)
     {
+        std::optional<CommittingBlockNumberTagger> committing_block_number_tagger;
+
+        if (storage.getSettings()->queue_mode)
+        {
+            auto partition_id = MergeTreePartition(current_block.partition).getID(metadata_snapshot->getPartitionKey().sample_block);
+
+            {
+                std::lock_guard guard(storage.committing_block_numbers_mutex);
+
+                Int64 block_number = storage.increment.get();
+                auto [_, inserted] = storage.committing_block_numbers[partition_id].insert(block_number);
+                chassert(inserted);
+
+                committing_block_number_tagger.emplace(
+                    storage.committing_block_numbers_mutex,
+                    storage.committing_block_numbers.at(partition_id),
```

Comment:
you could reuse `storage.committing_block_numbers.at(partition_id)` from line 121

Reply:
thanks, fixed